### DISCUSSION TITLE
Deferred CSS not working in checkout while critical css is enabled

### DISCUSSION THREAD

**Discussion by sandipansaha** [breezefront/community#322] [1. Help]
Hi @vovayatsyuk 

We have found an odd one. While we have enabled the "use_css_critical_path", in the checkout pages the deferred CSS files are not working. They are not being changed to media="print" to media="all" as expected. This is working on all the other pages except checkout.

URL: [URL]

Can you please suggest accordingly?

Thanks

**Comment by vovayatsyuk** [comment]
This is not a breeze code. It looks like you are using some third-party pagespeed module that adds onload attributes.

Firstly, breeze no longer uses `onload="..."`: https://github.com/breezefront/module-breeze/commit/a8ae95a4e6a63c4ce7811cf809fd981b9de1646f
Secondly, breeze never had `onload="this.onload=null"` code. We used `onload="media='all'"` previously.

**Reply by sandipansaha** [reply]
Thanks @vovayatsyuk for your reply.
However, I have found the codes in the breeze module itself and we are not using any other pagespeed module as well. Please correct me if I am wrong.

I am checking the latest module - swissup/module-breeze : v2.17.13

I am seeing this code in the plugin file which is being loaded into frontend.
vendor/swissup/module-breeze/Plugin/AsyncCssPlugin.php:58
vendor/swissup/module-breeze/etc/frontend/di.xml:6

I can see the code in GitHub repository as well.
https://github.com/breezefront/module-breeze/blob/master/Plugin/AsyncCssPlugin.php
https://github.com/breezefront/module-breeze/blob/master/etc/frontend/di.xml

Please let me know your thoughts!

Thank you

**Reply by vovayatsyuk** [reply]
Right.. I completely forgot about this plugin, sorry.

This plugin is a copy of the Magento plugin with fixed logic for Breeze. It looks like CSP module breaks lazy CSS on checkout (Luma checkout too).

Will work on the fix. Also, my previous commit is useless so I will revert it.

This PR fixes it in Magento: https://github.com/magento/magento2/pull/39040/files

**Reply by sandipansaha** [reply]
Great! Please let me know once pushed into packagist.
Thank you again for taking the time to look into it.

Thanks

**Reply by vovayatsyuk** [reply]
Done.
